db/migrate/20160916131013_create_connector_providers_configurations.rb

```
@@ -0,0 +1,37 @@
+Sequel.migration do
+  up do
+    create_table :connector_providers do
+      Uuid     :id, primary_key: true, default: 'uuid_generate_v4()'.lit
+      DateTime :created_at, default: Sequel::CURRENT_TIMESTAMP
+      DateTime :updated_at, default: Sequel::CURRENT_TIMESTAMP
+      String   :name, null: false
+    end
+
+    alter_table :connector_providers do
+      add_index :name
+    end
+
+    create_table :connector_configurations do
+      Uuid     :id, primary_key: true, default: 'uuid_generate_v4()'.lit
+      DateTime :created_at, default: Sequel::CURRENT_TIMESTAMP
+      DateTime :updated_at, default: Sequel::CURRENT_TIMESTAMP
+      Boolean  :enabled, null: false
+      Integer  :max_rows, null: true
+
+      foreign_key :user_id, :users, type: :uuid, null: true, on_delete: :cascade
+      foreign_key :organization_id, :organizations, type: :uuid, null: true, on_delete: :cascade
+      foreign_key :connector_provider_id, :connector_providers, type: :uuid, null: false, on_delete: :cascade
```

I feel cascade here might be a bit aggressive. I would prefer the deletion of a provider to fail if it's in use, rather than silently delete all configurations. We can still do it manually if we delete/deprecate a provider. Maybe being overcautious here, so up to you.